books_scrapper/books_scrapper/spiders/books.py

```diff
@@ -0,0 +1,81 @@
+from pathlib import Path
+
+import scrapy
+
+from scrapy.http import Response
+from scrapy.selector import Selector
+from selenium import webdriver
+from twisted.internet.defer import Deferred
+
+
+def extract_star_rating(class_name: str) -> int:
+    star_mapping = {
+        "One": 1,
+        "Two": 2,
+        "Three": 3,
+        "Four": 4,
+        "Five": 5
+    }
+    rating_verb = class_name.split()[-1]
+
+    if rating_verb in star_mapping:
+        return star_mapping[rating_verb]
+
+    return 0
+
+
+class BooksSpider(scrapy.Spider):
+    name = "books"
+    allowed_domains = ["books.toscrape.com"]
+    start_urls = ["https://books.toscrape.com/"]
+
+    def __init__(self, *args, **kwargs) -> None:
+        super().__init__(*args, **kwargs)
+        self.driver = webdriver.Chrome()
+
+    def close(self, reason: str) -> Deferred[None] | None:
+        self.driver.close()
+        return self.close(reason)
```

Comment:
The `close` method is incorrectly implemented, causing a recursion error. Instead of calling `self.close(reason)`, you should call `super().close(reason)` or handle the closure logic without recursion.